tests/test_IOB.py

```diff
@@ -0,0 +1,127 @@
+            "id": "002650eb-3d53-4a3d-b39f-6cd0c1ce58f2",
+            "normal": 3.5,
+            "subType": "normal",
+            "time": "2015-03-03T00:00:00.000Z",
+            "timezoneOffset": -480.0,
+            "type": "bolus",
+            "uploadId": "upid_abcdefghijklmnop"},
+            { "deviceId": "DemoData-123456789",
+            "deviceTime": "2015-03-03T00:20:00",
+            "id": "de34ecc6-841a-4e72-9f59-c3bcfb708feb",
+            "normal": 5.31,
+            "subType": "normal",
+            "time": "2015-03-03T00:20:00.000Z",
+            "timezoneOffset": -480.0,
+            "type": "bolus",
+            "uploadId": "upid_abcdefghijklmnop"}]
+        self.expected_output = [[1425340800, 3.5], [1425342000, 5.31]]
+        self.assertEqual(self.expected_output, insulin_on_board.format_bolus_for_wizard(self.input))
+   
+    def test_format_sqaure(self):
+        self.input = [{"deviceId": "DemoData-123456789",
+            "deviceTime": "2015-03-03T00:00:00",
+            "duration": 600000,
+            "extended": 2.8,
+            "id": "cdad40ee-9ec4-44bb-a466-afd1bf53e362",
+            "subType": "square",
+            "time": "2015-03-03T00:00:00.000Z",
+            "timezoneOffset": -480.0,
+            "type": "bolus",
+            "uploadId": "upid_abcdefghijklmnop"}]
+        self.expected_output = [[1425340800, 1.4],[1425341100, 1.4]]
+        self.assertEqual(self.expected_output, insulin_on_board.format_bolus_for_wizard(self.input))
+    
+    def test_format_dual_square(self):
+        self.input = [{"deviceId": "DemoData-123456789",
+            "deviceTime": "2015-03-03T00:00:00",
+            "duration": 600000,
+            "extended": 3.0,
+            "id": "002650eb-3d53-4a3d-b39f-6cd0c1ce58f2",
+            "normal": 4.0,
+            "subType": "dual/square",
+            "time": "2015-03-03T00:00:00.000Z",
+            "timezoneOffset": -480.0,
+            "type": "bolus",
+            "uploadId": "upid_abcdefghijklmnop"}]
+        self.expected_output = [[1425340800, 4.0], [1425340800, 1.5],  [1425341100, 1.5]]
+        self.assertEqual(self.expected_output, insulin_on_board.format_bolus_for_wizard(self.input))
+
+    def test_simple_iob_dict_creation(self):
+        self.input = [{"deviceId": "DemoData-123456789",
+            "deviceTime": "2015-03-03T00:00:00",
+            "id": "002650eb-3d53-4a3d-b39f-6cd0c1ce58f2",
+            "normal": 6,
+            "subType": "normal",
+            "time": "2015-03-03T00:00:00.000Z",
+            "timezoneOffset": -480.0,
+            "type": "bolus",
+            "uploadId": "upid_abcdefghijklmnop"}]
+        self.expected_output = {1425340800: 6.0, 1425341100: 5.5, 1425341400: 5.0, 
+                                1425341700: 4.5, 1425342000: 4.0, 1425342300: 3.5,
+                                1425342600: 3.0, 1425342900: 2.5, 1425343200: 2.0,
+                                1425343500: 1.5, 1425343800: 1.0, 1425344100: 0.5}
```

Comment:
This strategy for the expected inputs and outputs is almost impossible to read for someone (me) reading the tests because your inputs use ISO-formatted timestamps and your outputs use epoch times. It's OK to have code in your tests to make them more readable. (The other obvious alternative here is to have the dict use ISO timestamps for the keys; the choice between these alternatives comes down to the context of what's best for the rest of your non-test code.) Here the strategy I would use for better readability/understandability is to have a couple of assertions checking specific timestamps after the input bolus, for example:

```
res = insulin_on_board.create_iob_dict(input, action_time=1)
self.assertEqual(res[convertISOToEpoch('2015-03-03T00:30:00.000Z')], float(input[0]['normal']/2)) #tests that halfway through the action time, IOB is half the original bolus
```

You'll need to write the `convertISOToEpoch` function - and it might be the kind of thing you'll eventually want in some kind of test utilities/helpers module.